[DOCS-1700] Update docs for automation history

mdlinville opened the pull request:
## Description

[DOCS-1700] Update docs for automation history
- Add new page: View an automation's history
- Update existing automation pages with links to history documentation
- Add new page to docs.json navigation
- Convert Hugo shortcodes to Mintlify components
- Port from PR #1563 which was created during Hugo era

Ready to continue technical review. All comments from #1563 are addressed.

## Testing
- [x] Local build succeeds without errors
- [x] Local link check succeeds without errors
- [x] PR tests succeed



[DOCS-1700]: https://wandb.atlassian.net/browse/DOCS-1700?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
[DOCS-1700] Update docs for automation history

- Add new page: View an automation's history
- Update existing automation pages with links to history documentation
- Add new page to docs.json navigation
- Convert Hugo shortcodes to Mintlify components
- Port from PR #1563 which was created during Hugo era

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -12,27 +12,43 @@ This page describes _automations_ in W&B. [Create an automation](/models/automat
-- Filter runs by user or run name.
+- **Runs**: Trigger when a run meets or changes by a threshold on a specific metric. For example, trigger when a run's `loss` is less than 0.01 or when a run's `accuracy` improves by 5%.
+- **Artifacts**: Trigger when a new artifact version is created or when a specific alias is added to an artifact version. For example, trigger when a new model version is created or when the `production` alias is added to a model.
+- **Registries**: Trigger when a new model version is linked to a registry or when an alias is added to a model version in the registry. For example, trigger when a new model version is linked to the `Model Registry` or when the `production` alias is added to a model version in the registry.
```
**question:** Curious, is there a reason we decided to add separate bullet points for "Artifacts" vs "Registries" events?  This change seems to imply that the events themselves are qualitatively different categories -- as opposed to the same kinds of events (create artifact, link artifact, adding an alias, etc.) that can occur within a particular scope (collection, project, or registry).

Not strictly about automations history, but it seems like this could end up being misleading and/or noticeably inconsistent with:
- users' prior expectations, for those that already use automations
- the way automations are currently organized within both (a) the existing UI and (b) the automations API

mdlinville replied:
I think it is because Noa wanted to clarify that artifacts aren't stored in a registry but they are stored in a project, so registry automations can't be triggered on artifact creation, only linking. Did I misunderstand?

mdlinville replied:
PTAL, I simplified and restructured the Automation Events page to be less bifurcated, and also found some duplicated procedural instructions to remove since it's a reference page. Thanks for the suggestion.

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -12,27 +12,43 @@ This page describes _automations_ in W&B. [Create an automation](/models/automat
+### Webhook
+Call a webhook URL with a JSON payload containing information about the triggering event. This enables integration with external systems like CI/CD pipelines, model deployment services, or custom workflows. The webhook payload includes:
+- Event details (type, entity information)
+- User information
```
**issue:** IIRC this isn't accurate, could you elaborate on what this is referring to?  ("user information" is inherently ambiguous -- e.g. whether this refers to the user who last edited the automation, the user responsible for the triggering event, etc.  None of those are currently in the payload by default, but let me know if you had something else in mind)

mdlinville replied:
I'm not sure, I'll remove it and the timestamp.

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -12,27 +12,43 @@ This page describes _automations_ in W&B. [Create an automation](/models/automat
+Call a webhook URL with a JSON payload containing information about the triggering event. This enables integration with external systems like CI/CD pipelines, model deployment services, or custom workflows. The webhook payload includes:
+- Event details (type, entity information)
+- User information
+- Timestamp
```
**issue:** this is also not part of the payload

mdlinville replied:
```suggestion
```

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -12,27 +12,43 @@ This page describes _automations_ in W&B. [Create an automation](/models/automat
+- Event details (type, entity information)
+- User information
+- Timestamp
+- Custom payload data you configure
```
**issue:** the only payloads that webhooks support is this one (i.e. a custom, user-configured payload)

mdlinville replied:
```suggestion
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+---
+
+<Note>
+This feature is available only for W&B Enterprise Cloud customers.
```
**issue:** It's available on server 0.75+

mdlinville replied:
```suggestion
This feature requires an Enterprise license. It is available only for:
- W&B Multi-tenant Cloud
- W&B Dedicated Cloud
- W&B Self-Managed v0.75.0+
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+- **Success**: The automation completed all actions successfully.
+- **Failed**: The automation encountered an error and could not complete.
+- **Pending**: The automation is queued for execution.
+- **Running**: The automation is currently executing.
```
**note:** as previously mentioned -- 
- execution status can be `Finished`, `Failed`, or `Pending` (i.e. it uses the same names as run statuses for consistency).
- `Running` isn't currently an automation execution status, since we don't support any actions that run for long enough to capture this state

mdlinville replied:
```suggestion
- **Finished**: The automation completed all actions successfully.
- **Failed**: The automation encountered an error and could not complete.
- **Pending**: The automation is queued for execution.
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+1. **Trigger time**: The exact timestamp when the event occurred.
+1. **Execution time**: When the automation started processing.
+1. **Completion time**: When the automation finished (for completed executions).
+1. **Duration**: Total time taken to execute the automation.
```
**issue:** I think we mentioned this before, but we don't show duration or completion time in the UI (let me know if you see differently somewhere)

mdlinville replied:
```suggestion
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+
+- **Filter by status**: Show only successful, failed, pending, or running executions.
+- **Filter by date range**: View executions within a specific time period.
+- **Filter by automation**: Show executions for specific automations.
```
**issue:** filtering is ticketed, but definitely not supported in the UI at this time

mdlinville replied:
```suggestion
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+- **Filter by status**: Show only successful, failed, pending, or running executions.
+- **Filter by date range**: View executions within a specific time period.
+- **Filter by automation**: Show executions for specific automations.
+- **Search**: Search for executions by automation name, event type, or entity name.
```
**issue:** this sounds like it's still describing filtering on multiple attributes.  Right now search is only supported on automation name.

mdlinville replied:
```suggestion
To find an automation execution, you can search by automation name, or you can sort the list of executions by any column.
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+- **Authentication errors**: The webhook or Slack integration lacks proper credentials.
+- **Network errors**: The target endpoint is unreachable.
+- **Timeout**: The webhook took too long to respond.
+- **Invalid payload**: The webhook rejected the request due to malformed data.
```
**issue:** I'm not sure we can claim any of these.  The response and error returned by an arbitrary webhook request will depend entirely the webhook service itself, including how spec-compliant the webhook service is, how the server handles different types of errors (if at all), etc.

mdlinville replied:
```suggestion
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+
+- **History retention**: Execution history is retained for 90 days.
+- **Maximum executions**: Each automation can have up to 1000 executions in its history.
+- **Rate limiting**: Automations are subject to rate limits to prevent excessive executions.
```
**question:** Where are you seeing these?

mdlinville replied:
I'll remove them. They were present in the prior (Hugo) PR too, I think. The 90-day retention 
```suggestion
```

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+1. **Set up alerts**: Create automations to notify you when other automations fail.
+1. **Test thoroughly**: Use test events to verify automation behavior before deploying to production.
+1. **Document automations**: Include clear descriptions in your automations to make troubleshooting easier.
+1. **Clean up old automations**: Delete or disable automations that are no longer needed to reduce clutter in the history.
```
**note:** I would remove this section, as 
- It's potentially too prescriptive, as it assumes users only use automations for managing production workflows/pipelines
- It suggests support for features that don't yet exist, e.g. 
  - "create automations to notify you when other automations fail", or 
  - "use test events" which can imply proper support for mocking events or dry runs, depending on how it's interpreted)

mdlinville replied:
```suggestion
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+  - Response status (for webhooks)
+  - Any error messages or stack traces (for failed executions)
+
+## Filtering and searching
```
```suggestion
## Searching
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,153 @@
+1. For webhooks, confirm the endpoint is accessible and accepting requests.
+1. For Slack notifications, verify the channel exists and the bot has permission to post.
+
+## Retention and limits
```
```suggestion
```

mdlinville committed:
Tony's feedback

mdlinville committed:
Update models/automations/view-automation-history.mdx

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Restructure Automation Events page to be less bifurcated

- Consolidate artifact events
- Restructure the intro to use a matrix
- Remove some duplicated instructions that don't belong
  in a reference
- Remove some unneeded screenshots

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville commented on models/automations/automation-events.mdx:
```diff
@@ -7,81 +7,54 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
 </Note>
 
-Trigger a workflow when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. Optionally, you can further limit the runs that can trigger an automation by specifying a user or run name filter.
+In a project, an automation can start when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. An automation cannot start when a run's status is **Killed**.
```
```suggestion
In a project, an automation can start when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. An automation cannot start when a run's status is **Killed**. This status indicates that the run was stopped forcibly by an admin user.
```

mdlinville commented on models/automations/automation-events.mdx:
```diff
@@ -7,81 +7,54 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
-#### Run status change
+## Run status change
 <Note>
 - Currently available only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud).
 - A run with **Killed** status cannot trigger an automation. This status indicates that the run was stopped forcibly by an admin user.
```
```suggestion
Currently available only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud).
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+
+## View automation history
+
+You can view automation history from multiple locations in the W&B interface:
```
```suggestion
You can view automation history from multiple locations in the W&B App:
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+</Tab>
+<Tab title="Project">
+1. Navigate to your project from the W&B home page or by using the project selector.
+1. Click the **Automations** tab in the project navigation bar (located alongside Overview, Workspace, Runs, etc.). The project's automations display.
```
```suggestion
1. Click the **Automations** tab in the left navigation. The project's automations display.
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+1. In the **Automations history** tab, view all executions of the registry's automations in reverse chronological order, starting with the most recent execution. Each execution's metadata displays, including the event, action, and status.
+</Tab>
+<Tab title="Project">
+1. Navigate to your project from the W&B home page or by using the project selector.
```
```suggestion
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+
+    - Find the automation you want to investigate. You can use the search bar to filter by automation name, and you can sort by the last triggered date to find recently executed automations.
+
+    - Click an automation name to open its details page.
```
```suggestion
    - Click an automation's name to view or edit its configuration.
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+    - Find the automation you want to investigate. You can use the search bar to filter by automation name, and you can sort by the last triggered date to find recently executed automations.
+
+    - Click an automation name to open its details page.
+1. In the **History** tab, view all executions of the project's automations in reverse chronological order, starting with the most recent execution. Each execution's metadata displays, including the event, action, and status.
```
```suggestion
1. In the **History** tab, view all executions of the project's automations in reverse chronological order, starting with the most recent execution. Each execution's metadata displays, including the event, action, and status. Click an execution timestamp to view details about a particular execution.
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+
+1. Navigate to your project or registry.
+1. Click the **Automations** tab.
+1. Click on an automation name to open its details page.
```
```suggestion
1. Click an automation name to view or edit its configuration.
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+1. Navigate to your project or registry.
+1. Click the **Automations** tab.
+1. Click on an automation name to open its details page.
+1. The **Execution history** section shows all executions for this specific automation.
```
```suggestion
1. The **Execution history** section shows all executions for this specific automation. lick an execution timestamp to view details about a particular execution.
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,125 @@
+1. Check the event and action configuration.
+1. Verify that any required secrets are properly configured.
+1. For webhooks, confirm the endpoint is accessible and accepting requests.
+1. For Slack notifications, verify the channel exists and the bot has permission to post.
```
```suggestion
```

mdlinville committed:
Copyedits

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,111 @@
+
+
+
+
```
```suggestion
```

mdlinville committed:
Update models/automations/view-automation-history.mdx

mdlinville committed:
Remove reference to now deleted image

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Force public API filenames to lowercase

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,106 @@
+- W&B Self-Managed v0.75.0+
+</Note>
+
+This page describes how to view and understand the execution history of your W&B [automations](/models/automations) and track what triggered an automation, what actions were taken, and whether they succeeded or failed.
```
```suggestion
This page describes how to view and understand the execution history of your W&B [automations](/models/automations), including what triggered an automation, what actions were taken, and whether they succeeded or failed.
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,106 @@
+- **Pending**: The automation is queued for execution.
+
+
+### Detailed execution information
```
```suggestion
### Execution details
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,106 @@
+  - Response status (for webhooks)
+  - Any error messages or stack traces (for failed executions)
+
+## Searching
```
```suggestion
## Find an automation execution
```

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,106 @@
+## Searching
+
+To find an automation execution, you can search by automation name, or you can sort the list of executions by any column.
+## Next steps
```
```suggestion

## Next steps
```

mdlinville committed:
Copyedits

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -8,31 +8,43 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
-- When a metric for a run meets or exceeds a configured threshold.
-- When a metric for a run changes by a configured threshold.
-- When a run's status changes to **Running**, **Failed**, or **Finished**.
+Automations can be triggered by events related to runs or artifacts in a project or registry. An automation can start:
```
**suggestion:** 
```suggestion
Automations can be triggered by events related to runs or artifacts in a collection, project, or registry. An automation can start:
```

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -8,31 +8,43 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
-- Filter artifacts, aliases, or tags by name
-- Filter runs by user or run name.
+- When a run metric meets a defined absolute or relative threshold. For example, trigger when a run's `loss` is less than 0.01 or when a run's `accuracy` improves by 5%.
+- When an artifact event occurs in a project or registry. For example, trigger when a new model version is created or when the `production` alias is added to a model.
```
**suggestion:** The current text is technically accurate, and I definitely want to avoid describing features that **don't** exist now, but I think it's worth framing this less narrowly.  TL;DR I'd like to ensure we don't suggest or reinforce the impression that Automations is only ever meant for "artifacts and runs".

The fact that the non-run events are all related to artifacts is a historical accident.  Even within the current system, it wouldn't be too hard to add triggering events on other operations on W&B objects (e.g. "when user is added to a team / when tag is removed from a collection / when a registry is renamed / etc." -- it just happens these don't exist and/or aren't prioritized right _now_.)


```suggestion
- When an event occurs in a project or registry. For example, trigger when a new version of a model artifact is created or when the `production` alias is added to a model artifact.
```

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -8,31 +8,43 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
+When an event triggers an automation, it can perform one of these actions:
+
+### Slack notification
+Send a message to a Slack channel with details about the triggering event. The message includes relevant information such as the entity name, user who triggered the event, and a link to view more details in W&B.
```
**suggestion:** entity name and user aren't included in the default message.  It **is** possible to customize the message to include the entity/user -- by including those template variables in the custom message -- though I don't think we have specific examples of that in our docs yet.  This is expressly supported in the Automations API; need to remind myself if we currently allow this through the UI as well

mdlinville replied:
```suggestion
Send a message to a Slack channel with details about the triggering event. The message summarizes the event, with a link to view more details in W&B.
```

tonyyli-wandb commented on models/automations.mdx:
```diff
@@ -8,31 +8,43 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
+
+### Webhook
+Call a webhook URL with a JSON payload containing information about the triggering event. This enables integration with external systems like CI/CD pipelines, model deployment services, or custom workflows. The webhook payload includes:
+- Event details (type, entity information)
```
**suggestion:** the webhook payload is any JSON-serializable object and entirely up to the user -- it doesn't include anything by default
```suggestion
Call a webhook URL with a JSON payload containing information about the triggering event. This enables integration with external systems like CI/CD pipelines, model deployment services, or custom workflows. The body of the webhook request may be any JSON-serializable payload.
```

tonyyli-wandb commented on models/automations/automation-events.mdx:
```diff
@@ -7,81 +7,53 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
+| [Artifact version](#artifact-events) linked      | ✓              |               |
+| [Artifact version](#artifact-events) aliased     | ✓              | ✓             |
+| [Artifact version](#artifact-events) created     |                | ✓             |
+| [Artifact version](#artifact-events) tagged      |                | ✓             |
```
**suggestion:** "Artifact version created" is only supported in a collection scope (not project), so you may need to add a `Collection scope` column if you're keeping this change.

FWIW, you may also want to consider splitting the changes into at least 2 separate PRs, to unblock all the changes more easily:
- one for automations history edits
- one or more for other automation content edits

It's also worth noting these boundaries are more strictly curated in the UI.  Users who create/edit automations through the API will find that the table looks more like this:

```markdown
| Action                                           | Registry | Project | Collection 
|--------------------------------------------------|----------|---------|------------
| [Artifact version](#artifact-events) linked      | ✓        | ✓       | ✓
| [Artifact version](#artifact-events) aliased     | ✓        | ✓       | ✓
| [Artifact version](#artifact-events) created     |          |         | ✓
| [Artifact version](#artifact-events) tagged      | ✓        | ✓       | ✓
```

mdlinville replied:
This table was in response to feedback that the previous content was misleading because it was too bifurcated (maybe from the previous PR). I've adjusted these changes further, and split out to a separate PR.

tonyyli-wandb commented on models/automations/automation-events.mdx:
```diff
@@ -7,81 +7,53 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
 </Note>
 
-Trigger a workflow when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. Optionally, you can further limit the runs that can trigger an automation by specifying a user or run name filter.
+In a project, an automation can start when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. An automation cannot start when a run's status is **Killed**. This status indicates that the run was stopped forcibly by an admin user.
```
**question:** Have we confirmed the detail about `killed` status?  I'm actually not sure if `killed` is handled similarly to `crashed` by the backend (`run crashed` will trigger for any `run failed` event, but haven't been able to test the behavior for `run killed` yet)

mdlinville replied:
That fact is not new in this PR, it came with run status notifications docs, and was a correction from Stephen. https://github.com/wandb/docs/commit/0258b4373345b701ccf2ccae1c4dba296af30a73 It's looking like it's new because it is moved from elsewhere in the content. I suppose it could have changed in the meantime, but nobody has told me.

tonyyli-wandb reviewed (commented):
FWIW, you may want to consider splitting the changes into at least 2 separate PRs (or more as you see fit), to more easily unblock merging:

- one for Automations History edits (FWIW these LGTM to me now)
- at least one (or more) for edits on other Automations docs/content

mdlinville commented on models/automations/automation-events.mdx:
```diff
@@ -7,81 +7,53 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
 </Info>
 
-An automation can start when a specific event occurs within a project or registry. This page describes the events that can trigger an automation within each scope. Learn more about automations in the [Automations overview](/models/automations/) or [Create an automation](/models/automations/create-automations/).
+An automation can start when a specific event occurs within a project or registry. Learn more about automations in the [Automations overview](/models/automations/) or [Create an automation](/models/automations/create-automations/).
```
```suggestion
An automation starts when a specific event occurs within a project, registry, or collection. For example, when an artifact version is created, aliased, tagged, or linked, when a run's status changes, or when a run metric meets or changes by a given threshold. Learn more about automations in the [Automations overview](/models/automations/) or [Create an automation](/models/automations/create-automations/).
```

mdlinville commented on models/automations/automation-events.mdx:
```diff
@@ -7,81 +7,53 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
+| [Artifact version](#artifact-events) created     |                | ✓             |
+| [Artifact version](#artifact-events) tagged      |                | ✓             |
+| [Run status](#run-status-change) changed         |                | ✓             |
+| [Run metric](#run-metrics-change) threshold met  |                | ✓             |
```
```suggestion
```

mdlinville committed:
Apply suggestions from code review

Co-authored-by: Tony Li <tony.li@wandb.com>

mdlinville committed:
Merge remote-tracking branch 'origin/main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Further adjustments to events / scopes page

- Change Registry and Project H2s to tabs
- Put back specific details from before, revert the matrix
- Add regex support details to artifact alias and tag items
- Remove incorrect detail about artifact filters
- Remove the procedural instructions for getting to Registry
  or Project apps from this reference page
- Tabify Threshold and Change Threshold config details
- Disable the in-page TOC since it's not very useful

mdlinville committed:
Corrections

- Remove procedural instructions for getting to Registry and Project app, covered elsewhere
- Remove now-incorrect detail about artifact filters

mdlinville committed:
Merge remote-tracking branch 'origin/main' into cursor/update-docs-for-automation-history-81f2

mdlinville commented on models/automations.mdx:
```diff
@@ -8,31 +8,42 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
-- When a metric for a run meets or exceeds a configured threshold.
-- When a metric for a run changes by a configured threshold.
-- When a run's status changes to **Running**, **Failed**, or **Finished**.
+Automations can be triggered by events related to runs or artifacts in a collection, project, or registry. An automation can start:
```
```suggestion
Automations can be triggered by events related to runs or artifacts in a collection, project, or registry. For example, an automation can start:
```

mdlinville commented on models/automations.mdx:
```diff
@@ -8,31 +8,42 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
-- Filter artifacts, aliases, or tags by name
-- Filter runs by user or run name.
+- When a run metric meets a defined absolute or relative threshold. For example, trigger when a run's `loss` is less than 0.01 or when a run's `accuracy` improves by 5%.
+- When an event occurs in a project or registry. For example, trigger when a new version of a model artifact is created or when the `production` alias is added to a model artifact.
```
```suggestion
- When an artifact version is created, linked, or tagged in a project, registry, or collection. For example, trigger when a new version of a model artifact is created or when the `production` alias is added to a model artifact.
```

mdlinville commented on models/automations.mdx:
```diff
@@ -8,31 +8,42 @@ import EnterpriseCloudOnly from "/snippets/en/_includes/enterprise-cloud-only.md
-Optionally, you can:
-- Filter artifacts, aliases, or tags by name
-- Filter runs by user or run name.
+- When a run metric meets a defined absolute or relative threshold. For example, trigger when a run's `loss` is less than 0.01 or when a run's `accuracy` improves by 5%.
```
```suggestion
- When a run's status changes to **Running**, **Finished**, **Failed**.
- When a run metric meets a defined absolute or relative threshold. For example, trigger when a run's `loss` is less than 0.01 or when a run's `accuracy` improves by 5%.
```

mdlinville committed:
Reduce the detail in the overview

mdlinville committed:
Revert unrelated overview updates

mdlinville committed:
Merge remote-tracking branch 'origin/cursor/update-docs-for-automation-history-81f2' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

tonyyli-wandb commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,107 @@
+Each automation execution has one of the following statuses:
+
+- **Finished**: The automation completed all actions successfully.
+- **Failed**: The automation encountered an error and could not complete.
```
```suggestion
- **Failed**: The automation encountered an error and terminated unsuccessfully.
```

Technically the automation did "complete" (ie it exited without an _unhandled_ error), just unsuccessfully.  This contrasts with eg "crashed", which we don't support in automations right now, but that is a separate status on Runs (and automation status categories are deliberately a subset of run status categories)

tonyyli-wandb reviewed (approved):
Minor requested edit for precision/clarity, but LGTM otherwise, thanks!  Pre-approving to unblock, pending the requested change

mdlinville committed:
Update models/automations/view-automation-history.mdx

Co-authored-by: Tony Li <tony.li@wandb.com>

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

ngrayluna commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,107 @@
+
+    - Find the automation you want to investigate. You can use the search bar to filter by automation name, and you can sort by the last triggered date to find recently executed automations.
+
+    - Click an automation's name to view or edit its configuration.
```
Why are these nest bullet points? 
```suggestion
1. Find the automation you want to investigate. You can use the search bar to filter by automation name, and you can sort by the last triggered date to find recently executed automations.
1. Click an automation's name.
```

"... to view or edit its configuration." not sure why this is here, we are describing how to view automation history.

mdlinville replied:
Removed.

ngrayluna commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,107 @@
+
+You can view automation history from multiple locations in the W&B App:
+
+### From the Automations tab
```
Confused about this section title, the "From the automation details page" also requires clicking on a tab to view automation history. Suggest renaming sections. Or grouping "registry" vs "project" flow since that's how the product is split it.

mdlinville replied:
Refactored, thanks.

ngrayluna commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,107 @@
+  - Response status (for webhooks)
+  - Any error messages or stack traces (for failed executions)
+
+## Find an automation execution
```
Where are users supposed to search?

mdlinville replied:
In the search bar at the top of the execution history. Moved this info upward.

mdlinville committed:
Merge remote-tracking branch 'origin/main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge remote-tracking branch 'origin/cursor/update-docs-for-automation-history-81f2' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Refactored to address Noah's feedback, also add info and image about the automation execution summary in Registry collections

github-actions[bot] committed:
Optimised images with calibre/image-actions

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville committed:
Merge branch 'main' into cursor/update-docs-for-automation-history-81f2

mdlinville commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,66 @@
+1. View the registry's automations in the **Automations** tab. Click the **Last execution** timestamp to view the execution history details. You can use the search bar to filter by automation name, and you can sort by the last triggered date to find recently executed automations.
+1. View the registry's automation executions in reverse chronological order in the **Automations history** tab, including the event, action, and status. Click an execution timestamp to view details about a particular execution.
+
+<Tip>If a collection has associated automation executions, the icon ![Collection automation execution symbol, a circle with a right-pointing arrow](/images/automations/collection-automation-execution-icon) displays, along with the number of associated executions.</Tip>
```
```suggestion
<Tip>If a collection has associated automation executions, the icon ![Collection automation execution symbol, a circle with a right-pointing arrow](/images/automations/collection-automation-execution-icon.png) displays, along with the number of associated executions.</Tip>
```

mdlinville committed:
Update models/automations/view-automation-history.mdx

ngrayluna commented on models/automations/view-automation-history.mdx:
```diff
@@ -0,0 +1,66 @@
+- **Action details**: Information about what action was performed, such as notifying a Slack channel or running a webhook.
+- **Result details**: Additional information, if any, about the final outcome of executing the automation, including the error for a failed execution.
+
+Select a tab for details.
```
Nit: Add a one liner describing what the tabs are for. Something like "Based on your use case, select the **Registry** or **Project** tab to learn how to view your automation's history".

mdlinville replied:
```suggestion
Based on your use case, select the **Registry** or **Project** tab for detailed instructions for viewing automation history.
```

ngrayluna reviewed (commented):
Minor comment. Otherwise, looking good

mdlinville committed:
Update models/automations/view-automation-history.mdx
